corehq/middleware.py

```diff
@@ -210,3 +215,55 @@ def process_view(self, request, view_func, view_args, view_kwargs):
 
             if getattr(request, 'domain', None):
                 scope.set_tag('domain', request.domain)
+
+
+session_logger = logging.getLogger('session_access_log')
+log_file = os.path.join(settings.SHARED_DRIVE_ROOT, "session_access_log.txt")
+hadler = logging.FileHandler(log_file)
+session_logger.addHandler(hadler)
+session_logger.setLevel(logging.INFO)
+
+
+def log_call(func):
+    def with_logging(*args, **kwargs):
+        session_logger.info("\n\n\n")
+        session_logger.info(func.__name__ + " was called")
+        session_logger.info("\n\n\n")
+        for line in traceback.format_stack():
+            session_logger.info(line.strip())
+        return func(*args, **kwargs)
+    return with_logging
```

Comment:
this is going to output a massive amount of log data. Is there a way we can keep it to one line? Maybe if we just output the first line of the stackframe that hits our code?

Reply:
Won't be an issue since this will be enabled per domain toggle, right?

Reply:
I guess it depends on the domain's traffic